### ISSUE TITLE
Feature Request: Add option to reserve RAM

### ISSUE THREAD

**Issue by olia-dev** [vmware/terraform-provider-vcd#674]
Hello,

please check if it would be possible to add the option for RAM reservation to VMs (as can be configured in the webUI under the Advanced settings).
The VCD API also gives the option as `< rasd:Reservation > 0 </ rasd:Reservation >` https://code.vmware.com/apis/1159/vmware-cloud-director/doc/doc/operations/PUT-Memory-vApp.html

A VM defintion would then possible look like this:
```hcl
resource "vcd_vapp_vm" "vms" {
  name = "fqdn"

  memory = 4096
  memory_reservation = 4096
```

**Comment by Didainius** [comment]
I have checked these possibilities. It is possible but not using the current **PUT** `/api/vApp/vm-89c84bd6-c6f2-4e4c-8a7d-c44a3489e2e4/virtualHardwareSection/memory` endpoint. (this endpoint ignores change in `reservation` field)

Instead a `reconfigureVm` operation is required to change in `VM.VmSpecSection`. That way all `Advanced` tab in UI can be covered (including `limit`, `shares`, `reservation` and `priority`


I am leaving an experimental function to change CPU parameters (same would go for memory) in Go SDK. However this needs a bit more experimentation and checking if it handles well _hot_ and _cold_ updates (when they are enabled)
```go
func (vm *VM) ChangeCPUCountReservation(virtualCpuCount int, coresPerSocket *int, reservation int64) (Task, error) {
	if vm.VM.HREF == "" {
		return Task{}, fmt.Errorf("cannot update disks, VM HREF is unset")
	}

	//vmSpecSectionModified := true

	// Dereference vmSpectSection to avoid modifying parent
	newVmSpecSection := *vm.VM.VmSpecSection
	newVmSpecSection.Modified = takeBoolPointer(true)
	//cpuSpecSection := *vm.VM.VmSpecSection.CpuResourceMhz
	newVmSpecSection.CpuResourceMhz.Reservation = &reservation
	//newVmSpecSection.CpuResourceMhz.
	newVmSpecSection.NumCpus = takeIntAddress(virtualCpuCount)
	newVmSpecSection.NumCoresPerSocket = coresPerSocket

	//newVmSpecSection.CpuResourceMhz.Configured = int64(2)

	//disksSettingToUpdate.Modified = &vmSpecSectionModified

	return vm.client.ExecuteTaskRequest(vm.VM.HREF+"/action/reconfigureVm", http.MethodPost,
		types.MimeVM, "error updating CPU reservation: %s", &types.VMDiskChange{
			Xmlns:         types.XMLNamespaceVCloud,
			Ovf:           types.XMLNamespaceOVF,
			Name:          vm.VM.Name,
			VmSpecSection: &newVmSpecSection,
		})
}
````

**Comment by L3Nerd** [comment]
@Didainius You did the needed changes in Go SDK upstream?
I just stumbled over this feature because we need the CPU and memory reservation features for a project.

**Comment by vbauzysvmware** [comment]
Changes will be available in the next release 3.6.0
